library/src/main/java/com/getstream/sdk/chat/view/activity/AttachmentDocumentActivity.java

```diff
@@ -72,13 +72,19 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
 
         @Override
         public void onPageFinished(WebView view, String url) {
-            super.onPageFinished(view, StreamChat.getInstance(AttachmentDocumentActivity.this).getUploadStorage().signFileUrl(url));
+            if (view.getTitle().equals("")){
```

Comment:
Hm, that indeed works. Did you find it somewhere? Might be useful to leave here some comment why we're doing this and ideally link to SO answer description.

Reply:
Also what I'm afraid is that we may get here into infinite loop attempting to reload page. And in the worst case if it happens fast we will get blocked UI thread.

Lets add some counter and if it is reached like 5 reloads we show error message.

Reply:
> Hm, that indeed works. Did you find it somewhere? Might be useful to leave here some comment why we're doing this and ideally link to SO answer description.

I have found the solution from here.
https://stackoverflow.com/questions/53985055/pdf-sometimes-not-loading-with-google-embedded-viewer-on-android